src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Wpf.Authentication;
+using Greg.AuthProviders;
+using Reach;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.Publish.Models
+{
+    public class PublishModel
+    {
+        private readonly AuthenticationManager manager;
+        public LoginService LoginService;
+
+        private readonly string serverUrl;
+        private readonly string port;
+        private readonly string page;
+        private readonly string provider;
+
+        public bool IsLoggedIn
+        {
+            get
+            {
+                return manager.LoginState == LoginState.LoggedIn;
+            }
+        }
+
+        #region Initialization
+
+        public PublishModel()
+        {
+            // Open the configuration file using the dll location.
+            Configuration config = ConfigurationManager.OpenExeConfiguration(this.GetType().Assembly.Location);
+            // Get the appSettings section.
+            AppSettingsSection appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
+            serverUrl = appSettings.Settings["ServerUrl"].Value;
+            port = appSettings.Settings["Port"].Value;
+            page = appSettings.Settings["Page"].Value;
+            provider = appSettings.Settings["Provider"].Value;
```

Comment:
Let's make sure to check these for validity and throw an exception if they are null or empty.